pkg/driver/driver_openstack.go

```diff
@@ -468,3 +471,31 @@ func (d *OpenStackDriver) GetVolNames(specs []corev1.PersistentVolumeSpec) ([]st
 	}
 	return names, nil
 }
+
+func waitForStatus(c *gophercloud.ServiceClient, id string, pending []string, target []string, secs int) error {
+	return gophercloud.WaitFor(secs, func() (bool, error) {
+		current, err := servers.Get(c, id).Extract()
+		if err != nil {
+			return false, err
+		}
+
+		if strSliceContains(target, current.Status) {
+			return true, nil
+		}
+
+		if strSliceContains(pending, current.Status) {
+			return false, nil
+		}
+
+		return false, fmt.Errorf("unexpected status %q, wanted target %q", current.Status, strings.Join(target, ", "))
```

Comment:
Now for any server status that is not `BUILD` or `ACTIVE` we will return err. Isn't safer to check for `ERROR`? Also isn't possible for server to transition from `ERROR`/`UNKNOWN` to `ACTIVE`?
Just asking.

Reply:
I never saw the transition from `ERROR`/`UNKNOWN` to `ACTIVE`. I applied the [terraform](https://github.com/terraform-providers/terraform-provider-openstack/blob/8a6ba80eddd272885b6c600bbef9e4b433e1e421/openstack/resource_openstack_compute_instance_v2.go#L500..L507) logic, which showed that it works perfectly.

Reply:
Good to know that, thanks. Can you also update the PR with a release note describing the timeout increase and the new status handling?